torchtext/datasets/amazonreviewpolarity.py

```diff
@@ -25,20 +27,33 @@
     'test': f'{os.sep}'.join(['amazon_review_polarity_csv', 'test.csv']),
 }
 
-_EXTRACTED_FILES_MD5 = {
-    'train': "520937107c39a2d1d1f66cd410e9ed9e",
-    'test': "f4c8bded2ecbde5f996b675db6228f16"
-}
 
 DATASET_NAME = "AmazonReviewPolarity"
 
 
 @_add_docstring_header(num_lines=NUM_LINES, num_classes=2)
 @_create_dataset_directory(dataset_name=DATASET_NAME)
-@_wrap_split_argument(('train', 'test'))
+@_wrap_split_argument(("train", "test"))
 def AmazonReviewPolarity(root, split):
```

Comment:
Nit: do we want to take this as an opportunity to add typing info here?

Reply:
Agree on this one. Typing info and maybe comments on how the class is used would be quite useful and also consistent with our codebase.

Reply:
Added the annotations. The usage of datasets is common across all. As of now, we have added generic usage (https://pytorch.org/text/stable/datasets.html#torchtext-datasets, this need to be updated once we migrate to data pipes) for datasets that have only `root` and `split` as arguments. There are other datasets that have additional `kwargs` (example: https://pytorch.org/text/stable/datasets.html#iwslt2016) for which we may add additional doc-string examples.